src/bin/mdbook.rs

```diff
@@ -5,88 +5,52 @@ extern crate log;
-                        .arg_from_usage("--curly-quotes 'Convert straight quotes to curly quotes, except for those that occur in code blocks and code spans'")
-                        .arg_from_usage("[dir] 'A directory for your book{n}(Defaults to Current Directory when omitted)'"))
-                    .subcommand(SubCommand::with_name("serve")
-                        .about("Serve the book at http://localhost:3000. Rebuild and reload on change.")
-                        .arg_from_usage("[dir] 'A directory for your book{n}(Defaults to Current Directory when omitted)'")
-                        .arg_from_usage("-d, --dest-dir=[dest-dir] 'The output directory for your book{n}(Defaults to ./book when omitted)'")
-                        .arg_from_usage("--curly-quotes 'Convert straight quotes to curly quotes, except for those that occur in code blocks and code spans'")
-                        .arg_from_usage("-p, --port=[port] 'Use another port{n}(Defaults to 3000)'")
-                        .arg_from_usage("-w, --websocket-port=[ws-port] 'Use another port for the websocket connection (livereload){n}(Defaults to 3001)'")
-                        .arg_from_usage("-i, --interface=[interface] 'Interface to listen on{n}(Defaults to localhost)'")
-                        .arg_from_usage("-a, --address=[address] 'Address that the browser can reach the websocket server from{n}(Defaults to the interface address)'")
-                        .arg_from_usage("-o, --open 'Open the book server in a web browser'"))
-                    .subcommand(SubCommand::with_name("test")
-                        .about("Test that code samples compile"))
-                    .get_matches();
+    let app = App::new(NAME)
+                .about("Create a book in form of a static website from markdown files")
+                .author("Mathieu David <mathieudavid@mathieudavid.org>")
+                // Get the version from our Cargo.toml using clap's crate_version!() macro
+                .version(concat!("v",crate_version!()))
+                .setting(AppSettings::SubcommandRequired)
+                .after_help("For more information about a specific command, try `mdbook <command> --help`\nSource code for mdbook available at: https://github.com/azerupi/mdBook")
+                .subcommand(init::make_subcommand())
+                .subcommand(build::make_subcommand())
+                .subcommand(test::make_subcommand());
+
+    #[cfg(feature = "watch")]
+    let app = app.subcommand(watch::make_subcommand());
+    #[cfg(feature = "serve")]
+    let app = app.subcommand(serve::make_subcommand());
 
     // Check which subcomamnd the user ran...
-    let res = match matches.subcommand() {
-        ("init", Some(sub_matches)) => init(sub_matches),
-        ("build", Some(sub_matches)) => build(sub_matches),
+    let res = match app.get_matches().subcommand() {
+        ("init", Some(sub_matches)) => init::init(sub_matches),
```

Comment:
For consistency, what do you think of renaming the methods to `execute` or `execute_command`?
The name is already in the namespace so we wouldn't loose any information and it would make all the commands implement the same methods. :)

It's just a random note, I'm fine with the way it is now too.